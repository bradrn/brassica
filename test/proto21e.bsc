-x e / ə
-x g / ɣ
-x q / ʔ

new categories
Nasl = m  n   ?   ?  ŋ
Xstp = p  t   ?   ?  k  ʔ
Astp = ph th  ?   ?  kh
Nstp = mb nd  ?   ?  ng
Vstp = b  d
Xafr = ?  ts  tl  č
Aafr = ?  tsh tlh čh
Xfrc = f  s   lh  š  x  h
Vfrc = ?  ?   ?   ?  ɣ
Aprx = ?  ?   l   y  w

Stp = Xstp Astp Nstp Vstp
Afr = Xafr Aafr
Frc = Xfrc Vfrc

Obstr = Stp Afr Frc
Cont = Nasl Aprx

C = Nasl Stp Afr Frc Aprx

feature Vustr = æ a aa ɔ e ə o ii i u / Pstr = æʹ aʹ aaʹ ɔʹ eʹ əʹ oʹ iiʹ iʹ uʹ / Sstr = æ` a` aa` ɔ` e` ə` o` ii` i` u`

Str = Pstr Sstr

Vs = a ə i u
Vl = aa ii
V = Vs Vl

Other = ʹ
end

;-x Vustr / Sstr / Str ^Vustr C* _
;-x -rtl Vustr / Sstr / _ C* Vustr ^Str

; prenasalised → nasal
Nstp / Nasl

; [t] palatalisation
t / s / _ i

; deaffrication
Xafr / Xfrc / _

; aspirates → fricatives
Astp / Xfrc
Aafr / Xfrc

; glottal stops delete from beginning of syllable
; (except intervocalically)
ʔ / / # _
ʔ / / C _

; [ɣ] → [j] or [w]
ɣ / w / _ u
ɣ / y / _

; delete obstruents syllable-finally
Obstr / / _ [# C]

; pre-umlaut vowel shift
[a aa ə i ii u] / [æ ɔ ə e i u]

; stressed syllable umlaut
[i e æ ɔ u ə +Str] / [i e æ o u e +Str] / _ C* i
[i e æ ɔ u ə +Str] / [e æ æ ɔ o ə +Str] / _ C* e
[i e æ ɔ u ə +Str] / [e æ æ a o a +Str] / _ C* æ
[i e æ ɔ u ə +Str] / [e æ ɔ ɔ o ɔ +Str] / _ C* ɔ
[i e æ ɔ u ə +Str] / [i e ɔ o u o +Str] / _ C* u
[i e æ ɔ u ə +Str] / [e ə a ɔ o ə +Str] / _ C* ə

categories
V = æ a ɔ e ə o i u
end

; posttonic vowel syncope
V / / Str C* _

; stress clash resolution
; [V +Sstr] / [V +Vustr] / Str C* _
; [V +Sstr] / [V +Vustr] / _ C* Str

; [ɬ] → [š]
lh / š

new categories
Nasl = m  n  ɲ  ŋ
Xstp = p  t  č  k  ʔ
Vstp = b  d  j
Xfrc = f  s  š  x  h
Vfrc = ?  ?  ?  ɣ
Aprx = ?  l  y  w

Stp = Xstp Vstp
Frc = Xfrc Vfrc

Obstr = Stp Frc

C = Nasl Stp Frc Aprx

feature Vustr = æ a aa ɔ e ə o ii i u / Pstr = æʹ aʹ aaʹ ɔʹ eʹ əʹ oʹ iiʹ iʹ uʹ / Sstr = æ` a` aa` ɔ` e` ə` o` ii` i` u`

Str = Pstr Sstr

V = V = æ a ɔ e ə o i u
end

; [+stop] → [ʔ] before non-glottal stop + syllable end
Stp / ʔ / _ [Stp -ʔ] [C #]

; cluster metathesis and simplification:
; [j w ʔ] go next to vowel, [s] goes to end of syllable
C [y w] / \ / V _ [C #]
s C / \ / V _ [C #]
C ʔ / \ / V _

; [h]-deletion
-? h / / V _

; coronal/dorsal palatalisation
[n ŋ] y / ɲ
[k t] y / č
d y / j
[x s] y / š
k / č / _ [e i]

; nasal cluster smoothing
[m n ŋ l] / w / V _ [Nasl Aprx]

; diphthong smoothing
-? [w y] / / V _ C

; [a aa æ ɛ e ə ɔ o i ii u] ʹ / [á áá ǽ Ɛ é Ə 3 ó í íí ú] / _
; [a aa æ ɛ e ə ɔ o i ii u] ʹ / [á áá ǽ ɛ́ é ə́ ɔ́ ó í íí ú] / _
; ʹ / / _
