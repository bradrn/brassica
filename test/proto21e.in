sasat[wind] thaŋmu [DEF=SG] tlaqufli[follow-CONT] [sun]qalit thaŋ waqli gilut fawetlli [say[IMPF]-CONT]
naqeŋ khayiqenebey tshaal pawtiq
qaŋethinga siwitsi lamaq yusayetsi
ŋiiyusaye wiiletsi fawetli tsah naaqa waq qaŋeth thaŋ siifa lamaq qulu
ŋay nii khayiq
sasat thaŋ yusaa waq haŋuhaŋu qaŋeth thaŋ, kaday mahŋu ŋay
tsan naa qaŋeth thaŋ siwe lamaq thaŋ ŋandi mandaŋ phuum
qalit thaŋ yusayetsi siifitsi qaathanitsi waalhitsi qaŋeth thaŋ
qiyan paatliŋu qaŋeth thaŋ siifiŋu lamaq thaŋ qefayeŋu
sasat thaŋmu yusayeŋu wiiletsiŋu fawetliŋu
nikhayiqenedaa thaŋ tshaal bal