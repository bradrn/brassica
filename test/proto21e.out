SCA Output: sæsæ[wind] sæwmu [DEF=SG] šæʔule[follow-CONT] [sun]æle sæŋ wæle yelu fæwəle [say[IMPF]-CONT]
SCA Output: næʔəŋ xæyeʔənəbəy sɔl pæwse
SCA Output: æŋəseŋæ sewese læmæ yusæyəse
SCA Output: ŋiyusæyə wiləse fæwəše sæ nɔʔæ wæ æŋə sæŋ sifæ læmæ ulu
SCA Output: ŋæy ni xæye
SCA Output: sæsæ sæŋ yusɔ wæ hæŋuæŋu æŋə sæŋ� kædæy mæŋu ŋæy
SCA Output: sæn nɔ æŋə sæŋ sewə læmæ sæŋ ŋæne mænæŋ fuum
SCA Output: æle sæŋ yusæyəse sifese ɔsænese wɔšese æŋə sæŋ
SCA Output: eyæn pɔšeŋu æŋə sæŋ sifeŋu læmæ sæŋ əfæyəŋu
SCA Output: sæsæ sæwmu yusæyəŋu wiləseŋu fæwəšeŋu
SCA Output: nexæyeʔənədɔ sæŋ sɔl bæl
